library/keys.ml

```diff
@@ -92,8 +92,7 @@ let subst_keys (subst,(k,k')) =
   (subst_key subst k, subst_key subst k')
 
 let discharge_key = function
-  | KGlob g when Lib.is_in_section g ->
-    if isVarRef g then None else Some (KGlob (pop_global_reference g))
+  | KGlob (VarRef _ as g) when Lib.is_in_section g -> None
```

Comment:
Is there still a need for the `when`?

Reply:
I don't know, the logic is pretty mysterious to me, so I kept the behavior of the original code.